contracts/ts/genMaciState.ts

```diff
@@ -51,38 +53,70 @@ const genMaciStateFromContract = async (
     const stateTreeDepth = await maciContract.stateTreeDepth()
     assert(stateTreeDepth === maciState.stateTreeDepth)
 
-    // Fetch event logs
-    const initLogs = await provider.getLogs({
-        ...maciContract.filters.Init(),
-        fromBlock: fromBlock,
-    })
+    let initLogs: any[] = []
+    let signUpLogs: any[] = []
+    let mergeStateAqSubRootsLogs: any[] = []
+    let mergeStateAqLogs: any[] = []
+    let deployPollLogs: any[] = []
+    
+    const lastBlock = endBlock ? endBlock : await provider.getBlockNumber()
+    
+    // Fetch event logs in batches
+    for (let i = fromBlock; i < lastBlock; i += blocksPerRequest + 1) {
```

Comment:
I believe it's worthwhile to implement test code to avoid issues like [the previous case of duplicate logs](https://github.com/privacy-scaling-explorations/maci/issues/723#issuecomment-1771599471). Please note that such duplicate logs resulted in generating an incorrect proof.

Reply:
```
    // init() should only be called up to 1 time
    assert(
        initLogs.length <= 1, 
        'More than 1 init() event detected which should not be possible',
    )
```   
 If there are duplicate logs, this will throw.

Reply:
There are signUpLogs, mergeStateAqSubRootsLogs, mergeStateAqLogs, deployPollLogs, publishMessageLogs...etc. Without adequate tests, confidently predicting the impact of these changes on production becomes challenging. What are your thoughts on this?

Reply:
this code incorporates the fix needed to ensure we do not fetch duplicate logs - the same fix that the project encountering the issue proposed + We do have e2e and integration tests which use this genProof and have been working correctly.